avaje-config/src/main/java/io/avaje/config/InitialLoader.java

```diff
@@ -188,12 +190,29 @@ private boolean loadTest() {
   }
 
   /**
-   * Load configuration defined by a <em>load.properties</em> entry in properties file.
+   * Recursively Load configuration defined by a <em>load.properties</em> entry in properties file.
    */
   private void loadViaIndirection() {
     String paths = loadContext.indirectLocation();
     if (paths != null) {
-      loadViaPaths(paths);
+      var stack = new ArrayDeque<String>();
```

Comment:
Now that you have this as a loop and stack you can trivially add an integer counter of loaded resources.

If the amount exceeds lets say 64 we fail and we dump the contents of the array deque as circular culprints.